manifests/user.pp

```diff
@@ -126,10 +126,12 @@
     system         => $system,
   }
 
-  if $ensure == 'present' {
-    Group[$group] -> User[$name]
-  } else {
-    User[$name] -> Group[$group]
+  if $create_group {
+    if $ensure == 'present' {
+      Group[$name] -> User[$name]
```

Comment:
The assignment is different 

 Group[$group] -> User[$name] not  Group[$name] -> User[$name]